plugins/filetrees/yazi.nix

```diff
@@ -0,0 +1,170 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  types = lib.nixvim.nixvimTypes;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "yazi";
+  originalName = "yazi.nvim";
+  defaultPackage = pkgs.vimPlugins.yazi-nvim;
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  description = ''
+    Yazi is a blazing fast file manager for the terminal.
+    This plugin allows you to open yazi in a floating window in Neovim.
+
+    The default `<c-s>` grep function requires `telescope`.
+    You can override the grep functionality through the `plugins.yazi.settings.integrations.grep_in_directory`
+    and `plugins.yazi.settings.integrations.grep_in_selected_files` options
+    if you do not wish to use `telescope`.
+    Otherwise you must enable `plugins.telescope` or install it some other way to use this mapping.
+
+
+    The default `<c-g>` search and replace function requires `grug-far`.
+    You can override the replace functionality through the `plugins.yazi.settings.integrations.replace_in_directory`
+    and `plugins.yazi.settings.integrations.replace_in_selected_files` options
+    if you do not wish to use `grug-far`.
+    Otherwise you must install it to use this mapping.
```

Comment:
Would it be easier long term to just link to the upstream docs?

Something like

```md
Some default keybindings have additional dependencies you may need to install or enable.
See the [upstream docs](https://github.com/mikavilpas/yazi.nvim?tab=readme-ov-file#%EF%B8%8F-keybindings) for details.
```

Reply:
Yeah, fair enough. Less to maintain and keeps the ownership on upstream.